.github/workflows/e2e.yaml

```diff
@@ -29,7 +29,104 @@ on:
       circleCIToken:
         required: true
 jobs:
-  e2e:
+  e2e_gh_actions:
+    if: ${{ inputs.arch == 'amd64' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Print parameters"
+        run: |
+          echo "All inputs:"
+          echo "Running with: \
+              k8s: ${{ inputs.k8sVersion }} \
+              target: ${{ inputs.target }} \
+              parallelism: ${{ inputs.parallelism }} \
+              arch: ${{ inputs.arch }} \
+              cniNetworkPlugin: ${{ inputs.cniNetworkPlugin }} \
+              "
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.kuma-dev
+          key: ${{ runner.os }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-devtools
+      - uses: actions/download-artifact@v3
+        with:
+          name: build-output
+          path: build
+      - name: Install build tools
+        run: |
+          command: |
+            if ! command -v sudo 2>&1 >/dev/null; then
+              apt update
+              apt install -y sudo
+            fi
+
+            sudo apt update
+            sudo env DEBIAN_FRONTEND=noninteractive apt install -y curl git make unzip gcc xz-utils
```

Comment:
Do we need these? I think things like curl, git, make unzip are already installed no